src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        const index = Animal.alive.indexOf(target);
+
+        if (index !== -1) {
+          Animal.alive.splice(index, 1);
```

Comment:
Instead of manually checking the index and using `splice`, consider using the `filter` method to remove dead animals from the `Animal.alive` array. This approach is more concise and aligns with the checklist preference: 'It is preferred to filter animals with health less than or equal to 0 from `Animal.alive` array using Array's `filter` method.'